tests/test_multiplatform.py

```diff
@@ -0,0 +1,410 @@
+import os
+from typing import List
+from unittest.mock import patch
+
+import pytest
+from python_on_whales import docker
+from python_on_whales.exceptions import DockerException
+
+from buildrunner.docker.multiplatform_image_builder import (
+    ImageInfo, MultiplatformImageBuilder)
+
+TEST_DIR = os.path.basename(os.path.dirname(__file__))
+
+# FIXME: These tests can be broken if a custom buildx builder is set as default  # pylint: disable=fixme
+
+def actual_images_match_expected(actual_images, expected_images) -> List[str]:
+    missing_images = []
+    found = False
+    for expected_image in expected_images:
+        found = False
+        for actual_image in actual_images:
+            if actual_image.repo.endswith(expected_image):
+                found = True
+        if not found:
+            missing_images.append(expected_image)
+    return missing_images
+
+def test_use_local_registry():
+    registry_name = None
+    volume_name = None
+
+    with MultiplatformImageBuilder(use_local_registry=True) as mp:
+        registry_name = mp._reg_name
+
+        # Check that the registry is running and only one is found with that name
+        registry_container = docker.ps(filters={"name": registry_name})
+        assert len(registry_container) == 1
+        registry_container = registry_container[0]
+
+        # Check that the registry only has one mount
+        mounts = registry_container.mounts
+        assert len(mounts) == 1
+        mount = mounts[0]
+        assert mount.type == 'volume'
+        volume_name = mount.name
+        assert docker.volume.exists(volume_name)
+
+        # Check that the running container is the registry
+        assert registry_container.config.image == 'registry'
+        assert registry_container.state.running
+
+    # Check that the registry is stopped and cleaned up
+    registry_container = docker.ps(filters={"name": registry_name})
+    assert len(registry_container) == 0
+    assert not docker.volume.exists(volume_name)
+
+@pytest.mark.parametrize("name, in_mock_os, in_mock_arch, built_images, expected_image",
+                         [
```

Comment:
This indent seems a little much... but maybe it's just github doing weird things.

Reply:
No you are right, it is extreme. I'll make it less.

Reply:
done